src/ragapp/backend/agents/orchestrator.py

```diff
@@ -41,11 +41,17 @@ def get_agents(
             for tool_name, tool_config in agent_tools_config
             if tool_config.enabled
         ]
+        system_prompt = agent_config.get_system_prompt()
+        # Role is an information to introduce the agent to the orchestrator
+        # construct a custom "role" from the original role and goal for better orchestration
+        refined_agent_role = (
+            f"{agent_config.role}\n and its goals are {agent_config.goal}"
+        )
```

Comment:
```suggestion
        # The orchestrator uses "role" to select the right agent for a task
        # construct an "agent_selector" from the user defined role and goal for better orchestration
        agent_selector = (
            f"{agent_config.role}\n and its goals are {agent_config.goal}"
        )
```

Reply:
@leehuwuj This is a potential problem if the user doesn't select goal and role, the selection won't work

Suggestions:
1. we make role and goal required parameters
2. if the user selects a custom_prompt, he should also select the agent selector, defaults to "{role} and its goals are {goal}"

Reply:
BTW, isn't the original string: `The agent is an {role}`?